### PR TITLE
Add support for zx_channel_write_etc and zx_channel_read_etc.

### PR DESCRIPTION
**Pull request by chandarrengoog**
Add support for System.channelWriteEtc and System.channelQueryAndReadEtc.

These use the zx_channel_write_etc and zx_channel_read_etc syscalls, which implements the majority of https://fxbug.dev/68600.

Also add HandleDisposition, HandleInfoResult, and ReadEtcResult classes to dart:zircon's API surface.

I've added tests to channel_test and handle_disposition_test, and ran them locally.

Relevant bugs: https://fxbug.dev/68600

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/fuchsia/dart-pkg/zircon/lib/src/handle_disposition.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
```
Comment (arbreng) [inline shell/platform/fuchsia/dart-pkg/zircon/lib/src/handle_disposition.dart:1]: Please use Copyright 2013 here, I know it's strange but that's how we avoid adding a new license file and all that stuff

File: ci/licenses.sh [hunk 2]
```diff
@@ -131,7 +131,7 @@ function verify_licenses() (
       echo "It's very likely that this is an unintentional change. Please"
       echo "double-check that all newly added files have a BSD-style license"
       echo "header with the following copyright:"
-      echo "    Copyright 2013 The Flutter Authors. All rights reserved."
+      echo "    Copyright 2021 The Flutter Authors. All rights reserved."
```
Comment (arbreng) [inline ci/licenses.sh:134]: Please use Copyright 2013 in all places below, then you shouldn't need this

File: ci/licenses_golden/licenses_flutter [hunk 3]
```diff
@@ -2414,6 +2414,41 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ====================================================================================================
 
+====================================================================================================
```
Comment (arbreng) [inline ci/licenses_golden/licenses_flutter:2417]: Please use Copyright 2013 in all places below, then you shouldn't need this

File: shell/platform/fuchsia/dart-pkg/zircon/lib/src/handle_disposition.dart (hunk 1 above)
Comment (chandarrengoog) [reply shell/platform/fuchsia/dart-pkg/zircon/lib/src/handle_disposition.dart]: done.

File: ci/licenses_golden/licenses_flutter (hunk 3 above)
Comment (chandarrengoog) [reply ci/licenses_golden/licenses_flutter]: done.

**Review by chaselatta — APPROVED** [review]

**Pull request merged**